packages/backend/src/peripherals/database/SyncStatusRepository.ts

```diff
@@ -0,0 +1,27 @@
+import type { KeyValueStore } from './KeyValueStore'
+
+/** block of the first verifier deploy */
+const EARLIEST_BLOCK = 11813207
+
+export type Store = Pick<KeyValueStore<'lastBlockNumberSynced'>, 'get' | 'set'>
+
+export class SyncStatusRepository {
+  private readonly earliestBlock: number
+
+  constructor(
+    private readonly store: Store,
+    options: { earliestBlock?: number } = {}
+  ) {
+    this.earliestBlock = options.earliestBlock || EARLIEST_BLOCK
+  }
+
+  async getLastBlockNumberSynced(): Promise<number> {
+    const valueInDb = await this.store.get('lastBlockNumberSynced')
+
+    return (valueInDb && parseInt(valueInDb)) || this.earliestBlock
```

Comment:
I used `&&` `||` intentionally (maybe not very elegantly) to make sure this never fails, even if the kvStore contains a "corrupt value".

```
valueInDb = "database-is-corrupt?"

parseInt(valueInDb) // NaN
```

Also, when it's `0`, wouldn't we want to use `this.earliestBlock` instead?

Reply:
But it can be corrupted and still return a number, e.g: `parseInt('3 is my favorite number')` returns `3`. If we want to check this case we should have a check like `/^\d+$/.test(valueInDb)`

Reply:
We could go with `Number` instead of `parseInt`

```
Number('3 is my favorite number') // NaN
```

Reply:
Let's do `Number` then